src/drivers/magnetometer/hmc5883/hmc5883.cpp

```diff
@@ -989,7 +989,7 @@ HMC5883::collect()
 	sensor_is_onboard = !_interface->ioctl(MAGIOCGEXTERNAL, dummy);
 	new_report.is_external = !sensor_is_onboard;
 
-#ifdef PX4_I2C_BUS_ONBOARD_EXP
+#ifdef CONFIG_ARCH_BOARD_NXPHLITE_V3
```

Comment:
@DanielePettenuzzo I know this is a DNM HACK but just for the record we alway avoid any CONFIG_ARCH_BOARD_xxxx in code. common code. The thinking is name it as a feature and if def on that. [See](https://github.com/PX4/Firmware/pull/9392/commits/e0a551a51b87e959d686e2063d55d0ead958bfca)